services/channel/service/src/publishing/common/localization/get-validation-and-localizations.ts

```diff
@@ -0,0 +1,258 @@
+      const { data: publishData } =
+        await PrepareEntityLocalizationsForPublishing(
+          {
+            entityId,
+            entityType,
+            serviceId,
+            hash: data.validateEntityLocalizations.validationHash,
+          },
+          { Authorization: `Bearer ${authToken}` },
+        );
+      localizations =
+        publishData.prepareEntityLocalizationsForPublishing.localizations;
+    }
+
+    return {
+      localizations,
+      validation: data.validateEntityLocalizations,
+    };
+  } catch (e) {
+    throw getLocalizationMappedError(e);
+  }
+};
+
+export const getChannelValidationAndLocalizations = async (
+  localizationServiceBaseUrl: string,
+  authToken: string,
+  entityId: string,
+  entityType: string,
+  serviceId: string,
+): Promise<{
+  localizations: ChannelLocalization[] | undefined;
+  validations: PublishValidationMessage[];
+}> => {
+  try {
+    const { localizations, validation } = await getValidationAndLocalizations(
+      localizationServiceBaseUrl,
+      authToken,
+      entityId,
+      entityType,
+      serviceId,
+    );
+    return {
+      localizations: validateLocalizations(localizations)?.map((l) => ({
+        is_default_locale: l[LOCALIZATION_IS_DEFAULT_LOCALE],
+        language_tag: l[LOCALIZATION_LANGUAGE_TAG],
+        title: l.title,
+        description: l.description,
+      })),
+      validations: mapValidationMessages(
+        validation,
+        ({ locale, fieldName, message }: GqlValidationMessage) =>
+          `${locale}${fieldName ? `, ${fieldName}:` : ':'} ${message}`,
+      ),
+    };
+  } catch (e) {
+    throw getLocalizationMappedError(e);
+  }
+};
+
+export const getProgramValidationAndLocalizations = async (
+  localizationServiceBaseUrl: string,
+  authToken: string,
+  entityId: string,
+  entityType: string,
+  serviceId: string,
+  /**
+   * When playlist is published, we can get validation messages for multiple
+   * programs, so there is a need to identify which messages belong to which program.
+   */
+  programTitle?: string,
+): Promise<{
+  localizations?: ProgramLocalization[];
+  validations: PublishValidationMessage[];
+}> => {
+  try {
+    const { localizations, validation } = await getValidationAndLocalizations(
+      localizationServiceBaseUrl,
+      authToken,
+      entityId,
+      entityType,
+      serviceId,
+    );
+
+    return {
+      localizations: validateLocalizations(localizations)?.map((l) => ({
+        is_default_locale: l[LOCALIZATION_IS_DEFAULT_LOCALE],
+        language_tag: l[LOCALIZATION_LANGUAGE_TAG],
+        title: l.title,
+      })),
+      validations: mapValidationMessages(
+        validation,
+        ({ message }: GqlValidationMessage) =>
+          `${
+            programTitle
+              ? `The program "${programTitle}" is not fully localized (if the associated entity was localized those values will copied which may take a bit of time): `
```

Comment:
Minor adjustment needed here and in the test: `will copied` -> `will be copied` or `will be copied over`